### DISCUSSION TITLE
Authenticating against users in a SQL database - what are my options?

### DISCUSSION THREAD

**Discussion by jackdunncode** [goauthentik/authentik#8904] [Q&A]
Simply put; I have a SQL Server database with a Users table containing username and password.

What are my options of being able to use this as data source for Authentik to authenticate against?

FWIW, I already have a .NET Web API using ASP.NET Core Identity to manage and authenticate the users. But my client requires that the users authenticate via Authentik.

**Comment by BeryJu** [comment]
You should be able to accomplish this using policies. You could create a policy that is bound to the password stage of a login flow, and in the expression use the entered username to fetch things like the password hash from the database and save it to authentik, after which the password stage will authenticate against the saved hash. Depending on which SQL flavour this is (I'm assuming MS SQL) you'd most likely need a small bridge service that talks to MSSQL (since authentik doesn't ship with any MSSQL libraries) and serves an HTTP api, that you can talk to from the policy.
